src/app/Library/CrudPanel/Traits/Validation.php

```diff
@@ -123,30 +177,61 @@ public function getFormRequest()
     }
 
     /**
-     * Run the authorization and validation the currently set FormRequest.
+     * Run the authorization and validation for the current crud panel.
+     * That authorization is gathered from 3 places:
+     * - the FormRequest when provided.
+     * - the rules added in the controller.
+     * - the rules defined in the fields itself.
      *
      * @return \Illuminate\Http\Request
      */
     public function validateRequest()
     {
         $formRequest = $this->getFormRequest();
 
-        if ($formRequest) {
-            // authorize and validate the formRequest
-            // this is done automatically by Laravel's FormRequestServiceProvider
-            // because form requests implement ValidatesWhenResolved
-            $request = app($formRequest);
-        } else {
-            $request = $this->getRequest();
+        $_rules = $this->getOperationSetting('validationRules') ?? [];
+        $_messages = $this->getOperationSetting('validationMessages') ?? [];
```

Comment:
Why the underscore @pxpm ?

Reply:
I don't mind them _that_ much, but I'm just wondering if they're needed. We don't use this convention anywhere else.

Reply:
Refactored, I meant to use it only inside the anonymous class for distinction.